lib/services/data/budget/models/budget_item_ref_model.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:seren_ai_flutter/services/data/common/i_has_id.dart';
+import 'package:seren_ai_flutter/services/data/common/uuid.dart';
+
+part 'budget_item_ref_model.g.dart';
+
+@JsonSerializable()
+class BudgetItemRefModel implements IHasId {
+  @override
+  final String id;
+
+  @JsonKey(name: 'parent_org_id')
+  final String? parentOrgId;
+
+  final String type;
+  final String code;
+  final String source;
+  final String name;
+
+  @JsonKey(name: 'measure_unit')
+  final String measureUnit;
+
+  @JsonKey(name: 'base_unit_value')
+  final double baseUnitValue;
+
+  bool get isOwnSource => source == 'own';
+
+  BudgetItemRefModel({
+    String? id,
+    this.parentOrgId,
+    required this.type,
+    required this.code,
+    required this.source,
+    required this.name,
+    required this.measureUnit,
+    required this.baseUnitValue,
+  }) : id = id ?? uuid.v4();
+
+  /// Factory constructor to be used in loading states
+  factory BudgetItemRefModel.loading() => BudgetItemRefModel(
```

Comment:
Unsure why we have data state being intermixed with ui state. The UI should be responsible for showing the ... in the case of blank values.